java/arcs/core/data/plan.proto

```diff
@@ -0,0 +1,111 @@
+syntax = "proto3";
+
+package arcs;
+
+option java_package = "arcs.core.data.proto";
+option java_multiple_files = true;
+
+// Holds a single plan and particle specs used by it.
+message PlanEnvelope {
+  Plan plan = 1;
+  repeated ParticleSpec particle_specs = 2;
+}
+
+// Plan: A resolved recipe that can be instantiated.
+message Plan {
```

Comment:
FYI I opted to use "Proto" as a suffix to help disambiguate between the main class and the proto (VersionMap vs. VersionMapProto). I know you have this in a separate package, which means the symbols won't clash, but it does mean if you want to refer to both in the same file you will have to fully qualify one of them, which is annoying.

For anything that has a one-to-one correspondence with a Kotlin class (e.g. `Plan`), I'd recommend using the suffix (`PlanProto`).

Reply:
Good plan. There definitely will be places (like serialization and deserialization) where we will want to import both. Added Proto to all messages here.